pvlib/tests/test_floating.py

```diff
@@ -0,0 +1,33 @@
+import numpy as np
+import pandas as pd
+from pvlib import floating
+
+from .conftest import assert_series_equal
+from numpy.testing import assert_allclose
+
+
+def test_daily_stream_temperature_stefan_default():
+    result = floating.daily_stream_temperature_stefan(temp_air=15)
+    assert_allclose(result, 16.25, 0.01)
+
+
+def test_daily_stream_temperature_stefan_negative_temp():
+    result = floating.daily_stream_temperature_stefan(temp_air=-15)
+    assert_allclose(result, 0, 0.1)
+
+
+def test_daily_stream_temperature_stefan_ndarray():
+    air_temps = np.array([-5, 2.5, 20, 30, 40])
+    result = floating.daily_stream_temperature_stefan(temp_air=air_temps)
+    expected = np.array([1.25, 6.875, 20, 27.5, 35])
+    assert_allclose(expected, result, atol=1e-3)
+
+
+def test_daily_stream_temperature_stefan_series():
+    times = pd.date_range(start="2015-01-01 00:00", end="2015-01-05 00:00",
+                          freq="1d")
+    air_temps = pd.Series([-5, 2.5, 20, 30, 40], index=times)
```

Comment:
Both these lines plus the `expected = ...` down below and lines 20-21:

```python
    air_temps = np.array([-5, 2.5, 20, 30, 40])
    result = floating.daily_stream_temperature_stefan(temp_air=air_temps)
```

repeat the same values. You can try to use a fixture (a pytest.fixture). There are plenty of examples in the repo, for example this one:
https://github.com/pvlib/pvlib-python/blob/55e0a36fac1e3931b92f5f11508f6c09d903e0cc/pvlib/tests/test_irradiance.py#L68C1-L70C74

It would be great if you return both the input and the expected output from the function. [pytest.fixture docs](https://docs.pytest.org/en/7.1.x/how-to/fixtures.html).

Then, for the numpy types testing, use the numpy array view of the series, something along `pd.Series().values` or `.to_numpy()` (from the top of my head, if it gives you an error you won't have trouble finding what to use in the documentation).

Also, instead of testing the positive clipping of the output independently, you can add that to this data.